cpp/src/join/join.cu

```diff
@@ -13,32 +13,56 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include <cudf/table/table.hpp>
-
 #include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/dictionary/detail/update_keys.hpp>
 #include <cudf/join.hpp>
-#include <cudf/table/table_view.hpp>
+#include <cudf/table/table.hpp>
 
-#include "cudf/types.hpp"
-#include "hash_join.cuh"
+#include <join/hash_join.cuh>
 
 namespace cudf {
+namespace detail {
+namespace {
+
+cudf::table_view update_table_from_matched(cudf::table_view const& input,
+                                           cudf::table_view const& updated_table,
+                                           std::vector<size_type> const& indices)
+{
+  std::vector<cudf::column_view> updated_columns;
+  std::copy(input.begin(), input.end(), std::back_inserter(updated_columns));
+  // scatter( updated.begin(), updated.end(), indices.begin(), updated_columns.begin());
```

Comment:
Does a `thrust::scatter` with a `thrust::host` exec policy not work here?

Reply:
I tried that first but got a compile error about host-device calling mismatch somewhere. 
Do you know if there is a `scatter` pattern using `std` algorithms?

Reply:
Hm, I'm not sure how you'd do a scatter with a `std` algorithm.